src/QA/QA.Web/Orchestrators/DashboardOrchestrator.cs

```diff
@@ -14,52 +17,71 @@ public class DashboardOrchestrator
+            }
+
             if(string.IsNullOrEmpty(searchTerm)) return vm;
 
             vm.LastSearchTerm = searchTerm;
-            var result = await _vacancyClient.GetSearchResultsAsync(searchTerm);
-            vm.SearchResults = result.Select(v => MapToViewModel(v, vacancyUser)).ToList();
+            var searchResults = await _vacancyClient.GetSearchResultsAsync(searchTerm);
+            vm.SearchResults = searchResults.Select(v => MapToViewModel(v, vacancyUser)).ToList();
             return vm;
         }
 
-        private VacancyReviewSearchModel MapToViewModel(VacancyReviewSearch vacancyReviewSearch, VacancyUser vacancyUser)
+        private VacancyReviewSearchModel MapToViewModel(QaVacancySummary qaVacancySummary, VacancyUser vacancyUser)
         {
+            var userName = vacancyUser.UserId == qaVacancySummary.ReviewAssignedToUserId
+                ? "you"
+                : qaVacancySummary.ReviewAssignedToUserName;
+            var isAvailableForReview =
+                _vacancyClient.VacancyReviewCanBeAssigned(qaVacancySummary.Status, qaVacancySummary.ReviewStartedOn);
             return new VacancyReviewSearchModel()
             {
-                AssignedTo = vacancyReviewSearch.ReviewAssignedToUserName,
-                AssignedTimeElapsed = GetElapsedTime(vacancyReviewSearch.ReviewStartedOn),
-                ClosingDate = vacancyReviewSearch.ClosingDate.ToLocalTime(),
-                EmployerName = vacancyReviewSearch.EmployerName,
-                VacancyReference = $"VAC{vacancyReviewSearch.VacancyReference}",
-                VacancyTitle = vacancyReviewSearch.Title,
-                ReviewId = vacancyReviewSearch.Id,
-                SubmittedDate = vacancyReviewSearch.SubmittedDate.ToLocalTime(),
-                IsAvailableForReview = vacancyReviewSearch.ReviewAssignedToUserId == vacancyUser.UserId ||
-                                       _vacancyClient.VacancyReviewCanBeAssigned(vacancyReviewSearch.Status, vacancyReviewSearch.ReviewStartedOn)
+                
+                AssignmentInfo = isAvailableForReview ? string.Empty : GetAssignmentInfo(qaVacancySummary.ReviewStartedOn, userName),
+                ClosingDate = qaVacancySummary.ClosingDate.ToLocalTime(),
+                EmployerName = qaVacancySummary.EmployerName,
+                VacancyReference = $"VAC{qaVacancySummary.VacancyReference}",
+                VacancyTitle = qaVacancySummary.Title,
+                ReviewId = qaVacancySummary.Id,
+                SubmittedDate = qaVacancySummary.SubmittedDate.ToLocalTime(),
+                IsAvailableForReview = qaVacancySummary.ReviewAssignedToUserId == vacancyUser.UserId || isAvailableForReview,
+                IsNotAvailableForReview = !isAvailableForReview
             };
         }
 
-        private string GetElapsedTime(DateTime? value)
+        private string GetAssignmentInfo(DateTime? value, string userName)
```

Comment:
This method is now doing two things and is now more related to the view. Could we not leave the sentences in the view. Only the swappable parts could be worked out here or in the viewmodel. If the razor page isn't the place to put it, I would suggest a static utility method that is called from the razor page.

Reply:
I have consciously made the change to calculate the whole field in the orchestrator so I don't have to expose fields like username and timeelapsed which could now have value of "you" and "now" respectively.